lib/plugins.ts

```diff
@@ -436,11 +428,21 @@ export class Plugins {
    * @return {webdriver.promise.Promise|Q.Promise} A promise which resolves to the
    *     function's return value
    */
-  safeCallPluginFun(
-      pluginObj: ProtractorPlugin, funName: string, args: IArguments, promiseType: PromiseType,
-      failReturnVal: any): any {
-    var deferred = promiseType == PromiseType.Q ? Q.defer() : webdriver.promise.defer();
-    var logError = (e: any) => {
+  private safeCallPluginFun(
+      pluginObj: ProtractorPlugin, funName: string, args: any[], promiseType: PromiseType,
+      failReturnVal: any) {
+    let resultPromise: webdriver.promise.Promise<any>|q.Promise<any>;
```

Comment:
Why this way instead of my way?  Couldn't you have just given `deferred` the type `webdriver.promise.Deferred | q.Deferred`?